pageserver/src/page_service.rs

```diff
@@ -297,61 +257,49 @@ impl<'a> Read for CopyInReader<'a> {
 ///
 /// Main loop of the page service.
 ///
-/// Listens for connections, and launches a new handler thread for each.
+/// Listens for connections, and launches a new handler task for each.
 ///
-pub fn thread_main(
+pub async fn libpq_listener_main(
     conf: &'static PageServerConf,
     auth: Option<Arc<JwtAuth>>,
     listener: TcpListener,
     auth_type: AuthType,
 ) -> anyhow::Result<()> {
     listener.set_nonblocking(true)?;
```

Comment:
I wonder, if that's needed now.

Reply:
The `set_nonblocking(true)`? Yes, see https://docs.rs/tokio/latest/tokio/net/struct.TcpListener.html#method.from_std.